### PR TITLE
R6 - Update to new event system and some other changes

### PR DESCRIPTION
**Pull request by dotblank**
These are breaking changes and the mmo suite of plugins need to be updated.

### CODE REVIEW

File: src/main/java/mmo/Core/MMOCore.java [hunk 1]
```diff
@@ -315,9 +355,9 @@ public void onPlayerTeleport(PlayerTeleportEvent event) {
 		}
 	}
 
-	public class mmoCoreSpoutListener extends SpoutListener {
+	public class mmoCoreSpoutListener implements Listener {
```
Comment (Rycochet) [inline src/main/java/mmo/Core/MMOCore.java:323]: Can probably remove this bit and make it implement in the main class instead...

**Comment by Rycochet** [comment]
You've managed to split lines everywhere where it's not needed - no 80 column limit, so that needs to be fixed - apart from that I need to fix all the other plugins at the same time, but this one will get pulled as soon as I get time. Hopefully that's later today lol

**Comment by dotblank** [comment]
I removed some of the formatting goof-ups. My IDE played havoc with its auto formater.

File: src/main/java/mmo/Core/MMOCore.java (hunk 1 above)
Comment (dotblank) [inline src/main/java/mmo/Core/MMOCore.java:323]: Yea we can move it back into the main class but I think the plugin is still much more readable like this.

I generally do not like to keep it in the main plugin class unless the plugin is simple and compact.

Comment (Rycochet) [inline src/main/java/mmo/Core/MMOCore.java:323]: It's already a sub-class within the main class to keep it simple and compact lol

**Comment by Rycochet** [comment]
Still hundreds of needless formatting changes in MMOPlugin.java - just about to get some time for coding (I hope) lol

**Comment by Rycochet** [comment]
I should never say I have time when a certain someone might wake up... I don't have any time right now, and the typing is totally unrelated to coding... Honest...

**Pull request merged**